### PR TITLE
fix(encrypt): Fixed issues#5979

### LINKED ISSUE #5979: Data encryption produced an incorrect SQL when insert only values without cloumns
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
```xml
<dependency>
    <groupId>org.apache.shardingsphere</groupId>
    <artifactId>sharding-jdbc-core</artifactId>
    <version>4.1.1</version>
</dependency>
```
### Expected behavior
```sql
insert into t_my_users(id, username, password_enc, password, email) values (null, ?, ?, ?, ?) ::: [Tom2, uuSnOplnPkGpTRbSHgdNUQ==, 123456,tom@163.com]
```
### Actual behavior
```sql
insert into t_my_users(id, username, password_enc, password, email) values (null, ?, ?, ?, ?) ::: [Tom2, 123456, uuSnOplnPkGpTRbSHgdNUQ==, 123456]
```
Tom's email address is missing.
### Reason analyze (If you can)
Column and parameter indices are not aligned, which causes this problem.
### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
See the code following.
### Example codes for reproduce this issue (such as a github link).
```java

/*
DROP TABLE IF EXISTS `t_my_users`;

CREATE TABLE `t_my_users` (
	`id` INT (11) NOT NULL AUTO_INCREMENT,
	`username` VARCHAR (128) DEFAULT NULL,
	`password` VARCHAR (255) DEFAULT NULL,
	`password_enc` VARCHAR (255) DEFAULT NULL,
	`email` VARCHAR (24) DEFAULT NULL,
	PRIMARY KEY (`id`)
) ENGINE = INNODB AUTO_INCREMENT = 1 DEFAULT CHARSET = utf8;
 */
public class Main {

    public static void main(String[] args) throws SQLException {
        BasicDataSource basicDataSource = new BasicDataSource();
        basicDataSource.setDriverClassName("com.mysql.jdbc.Driver");
        basicDataSource.setUrl("jdbc:mysql://192.168.189.127:3306/test");
        basicDataSource.setUsername("root");
        basicDataSource.setPassword("root");

        // Configure the encrypt rule
        Properties props = new Properties();
        props.setProperty("aes.key.value", "123456");
        EncryptorRuleConfiguration encryptorConfig = new EncryptorRuleConfiguration("AES", props);
        EncryptColumnRuleConfiguration columnConfig = new EncryptColumnRuleConfiguration("password", "password_enc", "", "aes");
        EncryptTableRuleConfiguration tableConfig = new EncryptTableRuleConfiguration(Collections.singletonMap("password", columnConfig));
        EncryptRuleConfiguration encryptRuleConfig = new EncryptRuleConfiguration();
        encryptRuleConfig.getEncryptors().put("aes", encryptorConfig);
        encryptRuleConfig.getTables().put("t_my_users", tableConfig);
        Properties p = new Properties();
        p.setProperty("sql.show", "true");
        // Get data source
        DataSource dataSource = EncryptDataSourceFactory.createDataSource(basicDataSource, encryptRuleConfig, p);
        Connection conn = dataSource.getConnection();

        PreparedStatement statement1 = conn.prepareStatement("insert into t_my_users (username,password,email) values (?,?,?)");
        statement1.setString(1, "Tom1");
        statement1.setString(2, "123456");
        statement1.setString(3, "tom@163.com");
        statement1.execute();

        PreparedStatement statement2 = conn.prepareStatement("insert into t_my_users values (null,?,?,?)");
        statement2.setString(1, "Tom2");
        statement2.setString(2, "123456");
        statement2.setString(3, "tom@163.com");
        statement2.execute();  // Incorrect behavior.

        PreparedStatement statement3 = conn.prepareStatement("insert into t_my_users values (?,?,?,?)");
        statement3.setInt(1, 3);
        statement3.setString(2, "Tom3");
        statement3.setString(3, "123456");
        statement3.setString(4, "tom@163.com");
        statement3.execute(); 

        conn.close();
    }
}
```

### PR DESCRIPTION
**Pull request by DemonLms**
Fixed data encryption can not properly processing SQL mixed with `?` and literal.

Closes #5979

### CODE REVIEW

**Comment by tristaZero** [comment]
@DemonLms I am glad to see you start to involve in this community. Welcome!
It seems more efficient to have a talk about your PR to solve the related issue. So if it does not bother you, can we exchange WeChat number? Since I can not get your email, is it convenient for you to send your WeChat number to `panjuan@apache.org`? 
Looking forward to your response. :)

**Comment by DemonLms** [comment]
I apologize for my low-level BUG and irresponsible attitude, and I am reworking with this PR.

**Comment by DemonLms** [comment]
@tristaZero I seriously rewrite this PR and supplement some test cases. Please review again.

**Comment by DemonLms** [comment]
@tristaZero I have update the method implementation of `getParameterIndex`  using an existing.

File: shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-rewrite/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptInsertValuesTokenGenerator.java [hunk 1]
```diff
@@ -133,7 +133,8 @@ private void encryptToken(final InsertValue insertValueToken, final String table
     private void addPlainColumn(final InsertValue insertValueToken, final int columnIndex,
                                 final String tableName, final String columnName, final InsertValueContext insertValueContext, final Object originalValue) {
         if (getEncryptRule().findPlainColumn(tableName, columnName).isPresent()) {
-            DerivedSimpleExpressionSegment derivedExpressionSegment = insertValueContext.getParameters().isEmpty()
+            DerivedSimpleExpressionSegment derivedExpressionSegment = (insertValueContext.getParameters().isEmpty()
+                    || insertValueContext.getValueExpressions().get(columnIndex) instanceof LiteralExpressionSegment)
```
Comment (tristaZero) [inline shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-rewrite/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptInsertValuesTokenGenerator.java:137]: Hi, These codes appear twice, do you think it is better to exact a function for them?

Comment (DemonLms) [reply shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-rewrite/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/impl/EncryptInsertValuesTokenGenerator.java]: Done.

**Comment by tristaZero** [comment]
@DemonLms Hi, Congratulations on your first one to Apache ShardingSphere community!
It is great PR, which helps users have better usage of Encryption feature.

**Pull request merged**